Zotero/Scenes/Detail/PDF/Views/PDFDocumentViewController.swift

```diff
@@ -428,6 +433,43 @@ final class PDFDocumentViewController: UIViewController {
 
     // MARK: - Selection
 
+    /// Shows temporary preview highlight in given rects. Used by note editor to highlight original position of annotation. The annotation may already be deleted, so we're highlighting the original location.
+    /// - parameter previewRects: Rects to select.
+    /// - parameter pageIndex: Page index of page where (selection should happen.
+    /// - parameter document: Active `Document` instance.
+    private func show(previewRects: [CGRect], pageIndex: PageIndex, document: PSPDFKit.Document) {
+        guard !previewRects.isEmpty, let pageView = self.pdfController?.pageViewForPage(at: pageIndex) else { return }
+
+        let convertedRects = previewRects.map({ pageView.convert($0, from: pageView.pdfCoordinateSpace) })
+        let view = AnnotationPreviewView(frames: convertedRects)
+        view.alpha = 0
+        pageView.contentView.addSubview(view)
+
+        UIView.animate(
```

Comment:
Instead of using `asyncAfter` above and below, maybe the `animated` parameters `delay` and `options: .curveEaseInOut` can be used for simplicity?

Reply:
The initial show preview function call still needs to be wrapped in `DispatchQueue.main.async` though, the initial state change is called too soon and without the delay the preview just doesn't show up - 32da1b737a859c087e1248a9bb1c65b4301665a6